src/2sv/codes/New.vue

```diff
@@ -118,51 +121,62 @@
 <script>
 import ProfileWizard from '@/profile/ProfileWizard.vue'
 import { add } from '@/global/mfa'
+import { formatLongDate } from '@/global/filters'
+import { useDisplay } from 'vuetify'
 
 export default {
+  name: 'NewCodes',
   components: {
     ProfileWizard,
   },
-  data: () => ({
-    codes: [],
-    printing: false,
-    copied: false,
-    gotEm: false,
-  }),
-  async created() {
-    const newCodes = await add('backupcode')
-
-    this.codes = newCodes.data
-
-    this.$refs.wizard.completed()
+  data() {
+    return {
+      codes: [],
+      printing: false,
+      copied: false,
+      gotEm: false,
+    }
   },
   computed: {
+    idpConfig() {
+      return this.idpConfig
+    },
     encodedData() {
-      return encodeURIComponent(`${this.$root.idpConfig.idpName}\r\n${this.codes.join('\r\n')}`)
+      return encodeURIComponent(`${this.$idpConfig.idpName}\r\n${this.codes.join('\r\n')}`)
     },
     mobile() {
-      return this.$vuetify.breakpoint.name === 'xs'
+      const { xs } = useDisplay()
+      return xs.value
     },
   },
+  async created() {
+    const newCodes = await add('backupcode')
+    this.codes = newCodes.data
+    this.$refs.wizard.completed()
+  },
   methods: {
-    finish: function () {
+    formatLongDate,
+    finish() {
       this.$refs.wizard.next()
     },
-    print: async function (id) {
-      this.gotEm = this.printing = true
+    async print(id) {
+      this.printing = true
+      this.gotEm = this.printing === true
       const el = document.querySelector(id)
 
       el.classList.add('printable')
 
-      // don't print until the renderer has run again with the newly added 'printable' class.
+      // Wait for DOM updates
       await this.$nextTick()
+
       window.print()
+
       this.printing = false
     },
-    copy: async function () {
-      await navigator.clipboard.writeText(`${this.$root.idpConfig.idpName}\r\n${this.codes.join('\r\n')}`)
-
-      this.gotEm = this.copied = true
+    async copy() {
+      await navigator.clipboard.writeText(`${this.$idpConfig.idpName}\r\n${this.codes.join('\r\n')}`)
+      this.copied = true
+      this.gotEm = this.copied === true
```

Comment:
Isn't this always going to set `this.gotEm` to true?

Reply:
yes, there was a bug and I after fixing I didn't get around to cleaning this up